SpatialGDK/Documentation/content/singleton-actors.md

```diff
@@ -1,55 +1,84 @@
+     ```
+     UCLASS(SpatialType=(Singleton, ServerOnly))
+     class AScoreBoard : public AActor
+     {
+       GENERATED_BODY()
+       ...
+     }
+     ``` 
 
-```
-UCLASS(SpatialType=(Singleton, ServerOnly))
-class TESTSUITE_API AExampleGameGameMode : public AGameModeBase
-{
-  GENERATED_BODY()
-  ...
-}
-```
+### Unreal Blueprint Class
 
-To tag a Blueprint class as a Public Singleton, open the class in the Blueprint Editor and navigate to the `Class Settings`. In the `Advanced` section inside `Class Options`, check the `Spatial Type` checkbox and add `Singleton` to the `Spatial Description` textbox. To tag a Blueprint class as a Private Singleton, follow the same steps and add `ServerOnly` to the `Spatial Description` textbox.
+1. Open your Blueprint Class in the [Blueprint Editor](https://docs.unrealengine.com/en-us/Engine/Blueprints/Editor).
+2. From the [Toolbar](https://docs.unrealengine.com/en-US/Engine/Blueprints/Editor/UIComponents/Toolbar), click **Class Settings** to open Blueprint Properties in the Default pane.
+3. In the **Class Options** section, ensure that the **Spatial Type** check box is selected.
+4. In the **Spatial Description** text box, tag the Blueprint Class with the relevant attribute(s), depending on what type of Singleton Actor you want it to be:
+   * **Public Singleton Actor**: enter `Singleton`.
+   * **Private Singleton Actor**: enter `Singleton ServerOnly`.
+    
+    Here's a screenshot of what your Blueprint Class looks like if you tag it as a Private Singleton Actor:
+    ![Singleton Blueprint]({{assetRoot}}assets/screen-grabs/blueprint-singleton.png)
 
-This is an example of what your Blueprint `Class Options` should look like if you've tagged it as a Private Singleton:
-![Singleton Blueprint]({{assetRoot}}assets/screen-grabs/blueprint-singleton.png)
+## Spawning Singleton Actors
 
-And that's it! You have successfully specified a Singleton Actor. Make sure that if you edit your Actor, you then generate [schema]({{urlRoot}}/content/spatialos-concepts/schema-and-snapshots#schema) and create a new [snapshot]({{urlRoot}}/content/how-to-use-snapshots) using the [SpatialOS GDK toolbar]({{urlRoot}}/content/toolbars).
\ No newline at end of file
+For all the following classes, which are automatically tagged as Singleton Actors, Singleton Actors are spawned automatically on all server-worker instances as part of the [Unreal Engine game flow](https://docs.unrealengine.com/en-US/Gameplay/Framework/GameFlow):
+
+  - `AGameMode`
+  - `AGameState`
+  - `ALevelScriptActor`
+  - any classes that derive from the preceding classes
+
+For all classes that you manually tag as Singleton Actors, make sure that Singleton Actors are spawned on all server-worker instances at runtime. You need to write code to do this when you create each server-worker type. 
```

Comment:
I'm not sure the second sentence is necessary. Spawning Singleton Actors needs to be done independently of the server-worker type, and we don't even official support other server-worker types currently.

Reply:
Ah, I thought the spawning bit was some code you had to write as part of the code for setting up your worker type. But I don't have a good mental model of this, having never done it myself. How do you actually get (manually tagged) Singleton Actors to spawn on each worker instance?

Reply:
You'd spawn them like you would any normal actor. Most likely it would be part of the game mode's begin play logic. It's entirely up to the user. I think this is covered in the first sentence though.

Reply:
OK, cool - as well as removing this sentence, would it make sense to replace

> **Note**: Never write code to spawn Singleton Actors when you create client-worker types.

with

> **Note**: Never spawn Singleton Actors on client-worker instances.

?

(To me, this sounds strange, because the user isn't directly doing the spawning - they're writing the code to do the spawning - but perhaps in context it makes sense?)

Reply:
Yes sounds good.

Reply:
Since it's not *write code to worker types*, I think we also need to change the following sentence from:
> [Spawn Singleton Actors](#spawning-singleton-actors): Write code to spawn Singleton Actors on server-worker instances at runtime. 

possibly to

> [Spawn Singleton Actors](#spawning-singleton-actors): Understand how Singleton Actors are spawned on server-worker instances at runtime.

Reply:
I think this update makes sense 👍